sympy/functions/combinatorial/factorials.py

```diff
@@ -866,192 +959,505 @@ def fdiff(self, argindex=1):
+                            return True
+                        elif n.is_extended_positive:
+                            return False
+                    elif n.is_finite:
+                        return True
+        elif k.is_real and k.is_integer is False and (
+                n - k).is_integer is False:
             return False
 
     def _eval_is_nonnegative(self):
         n, k = self.args
-        if n.is_integer and k.is_integer:
-            if n.is_nonnegative or k.is_negative or k.is_even:
-                return True
-            elif k.is_even is False:
-                return  False
+        if (n - k).is_zero or k.is_zero or n.is_nonnegative:
+            return True
+        if n.is_negative:
+            if k.is_negative:
+                if (n - k).is_positive:
+                    if (k.is_even == n.is_even) and k.is_even is not None:
+                        return True
+                    elif (k.is_even != n.is_even) and (
+                        k.is_even is not None) and (n.is_even is not None):
+                        return False
+                elif (n - k).is_nonpositive:
+                    return True
+            elif k.is_positive:
+                if k.is_odd:
+                    return False
+                elif k.is_even:
+                    return True
+
+
+class multinomial(CombinatorialFunction):
+    r"""Implementation of the multinomial coefficient, a
+    generalization of binomial coefficients. They can be expressed in terms
+    of binomials, factorials or the gamma function:
+
+    .. math:: \frac{(k_1+k_2+\cdots+k_m)!}{k_1!\, k_2! \cdots k_m!} = {k_1\choose k_1}{k_1+k_2\choose k_2}\cdots{k_1+k_2+\cdots+k_m\choose k_m}
+
+    The multinomial coefficients have a direct combinatorial interpretation,
+    as the number of ways of depositing ``n`` distinct objects into ``m`` bins,
+    with ``k_i`` objects in a given bin.
+
+    For non-integral values of ``k`` which sum to ``n``, the multinomial gives the
+    coefficient of the terms in the series expansion of ``(a_1 + a_2 + ... + a_m) ^ n`` where
+    ``k_i`` is the exponents of ``a_i`` in a term of the expansion.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, Rational, multinomial, expand_func, symbols, gamma
+
+    Consider the following polynomial in three variables:
+
+    >>> a, b, c = symbols('a b c')
+    >>> ((a + b + c)**3).expand()
+    a**3 + 3*a**2*b + 3*a**2*c + 3*a*b**2 + 6*a*b*c + 3*a*c**2 + b**3 + 3*b**2*c + 3*b*c**2 + c**3
+
+    The coefficient for any term with variables having exponents of 0, 1 and 2 is 3:
+
+    >>> multinomial(0, 1, 2)  # order does not matter
+    3
+
+    The series expansion of `(1 + x)^(2/3)` yields the following leading terms when it
+    is expanded:
+    ``1 + (2/3) x^1 - (1/9) x^2 + (4/81) x^3 - (7/243) x^4 + ...``. The coeffient of
+    ``x^2`` is given by ``multinomial(2, a)`` where (since the arguments must sum to 2/3)
+    ``a = -4/3``.
```

Comment:
```suggestion
    $1 + (2/3) x^1 - (1/9) x^2 + (4/81) x^3 - (7/243) x^4 + ...$. The coefficient of
    $x^2$ is given by ``multinomial(2, a)`` where (since the arguments must sum to 2/3)
    $a = -4/3$.
```